### DISCUSSION TITLE
An alternative to `c.prefix.tree` from Scala2 reflect

### DISCUSSION THREAD

**Discussion by soronpo** [scala/scala3#12470] [Metaprogramming]
I'm looking for an equivalent to Scala 2's `c.prefix.tree`.
Is there such a thing? E.g.:
```scala
class Foo(int i)(using Macro)
val x = new Foo(5)
```
Inside the macro I'll be able to retrieve the tree `new Foo(5)` (I don't want the text, but the actual tree)

**Comment by nicolasstucki** [comment]
What is this used for? 

How is it used?

What does it give?

Can it break the signature guarantees?
